extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java

```diff
@@ -128,6 +133,77 @@ public class InfinispanClientRuntimeConfig {
     @ConfigItem
     Optional<String> trustStoreType;
 
+    /**
+     * Configures caches from the client with the provided configuration.
+     */
+    @ConfigItem
+    Map<String, RemoteCacheConfig> cache = new HashMap<>();
+
+    @ConfigGroup
+    public static class RemoteCacheConfig {
+
+        // @formatter:off
+        /**
+         * Cache configuration in inlined XML to create the cache on first access.
+         * Will be ignored if the configuration-file or the configuration-uri is provided for the same cache name.
+         * An example of the user defined property:
+         * quarkus.infinispan-client.cache.bookscache.configuration=<distributed-cache><encoding media-type="application/x-protostream"/></distributed-cache>
+         */
+        // @formatter:on
+        @ConfigItem
+        public Optional<String> configuration;
+
+        // @formatter:off
+        /**
+         * Cache configuration file in XML whose content will be used to create the cache on first access.
+         * Will be ignored if configuration-uri is provided.
+         *
+         * An example of the user defined property:
+         * quarkus.infinispan-client.cache.bookscache.configuration-file=cacheConfig.xml
+         */
+        // @formatter:on
+        @ConfigItem
+        public Optional<String> configurationFile;
```

Comment:
If neither is set, shouldn't we make the default protostream? Or is it already on the server side?

Reply:
Protostream is the default marshaller on client side. Caches configuration must define the encoding in the configuration when they are created in server side

Reply:
That's my point exactly: if protostream is the default for the client, shouldn't it be the default for the caches we create through configuration?

My point is that if no configuration is provided at all, I would have created the cache with a protostream encoding.

That's just a proposal though, I'm not very knowing Infinispan well enough :).

Reply:
@karesti I'm not sure why this conversation was solved but it's still a problem IMO. Basically, if you want things to work for Quarkus default setup, you will have to define a configuration to enforce protostream. See https://github.com/quarkusio/quarkus-quickstarts/pull/1156/files#diff-49609e77c8938a804c7f8ebf93619dc1f0822e26bf85a2e65e7b14c834b0150aR1 for instance.

Wondering how we could improve the situation? And maybe push that default config if everything is empty?

Reply:
@gsmet I resolved the issue since I thought I answered the question already. The Infinispan Client in Quarkus uses the protostream marshaller. You can use a cache that does not configure protostream encoding explicitly with the infinispan client using protostream marshalling (text cache for example should work well). Moreover, indexed caches, without specifying any encoding, get protostream encoding for granted. I understand this is confusing since it touches infinispan cache configuration specifics and this confusion/complexity is one of the main reasons the cache configuration wizard is introduced in Infinispan 14. Users are free to use any cache configuration they need, even if the recommended encoding for Infinispan today is protostream. 

The configuration provided by these tests is not a generic default configuration of a cache. Most of the time Infinispan caches configuration is far way more detailed in other aspects: number of owners, RBAC, persistence ... 
That said, I see the point of providing a default for non existing caches for Quarkus specially in dev mode and tests. Definitely that will smooth the experience for an Infinispan newby. With the cache wizard feature that will be also highlighted in Quarkus when I will update to Infinispan 14, the experience should also be improved. 

The problem I see today is that the RemoteCacheManager is created using the application.properties before any injection of a cache is done. Configuring caches like we do in this PR, creates the caches on first access, so during the injection. If we don't configure anything, and we provide a name in the remote cache, we don't have that name before the remote cache manager is created. What we could maybe do is using the administration API to create the cache in the getRemoteCache method in the Infinispan Client Producer. Then we inject the RemoteCache and the name is in the Remote Annotation.  

I just need to talk to the team to check this enhancement for Quarkus to implement this.

How does this sound ?

Reply:
After discussion, I will implement a way to create those caches on the fly if they don't exist with the provided small config and protobuf.
When I will update to 14 more things around will come 
https://issues.redhat.com/browse/ISPN-14124